### PR TITLE
Added awesome complete support.

### LINKED ISSUE #298: Support for addon Github Awesome Autocomplete
### Feature Request
Add support of the addon [Github Awesome Autocomplete](https://github.com/algolia/github-awesome-autocomplete).

Doesn't look great currently:
picture

Cross-post from [here](https://github.com/algolia/github-awesome-autocomplete/issues/34)

### PR DESCRIPTION
**Pull request by rayrutjes**
Preview:
image

Fixes #298, algolia/github-awesome-autocomplete#34

Let me know what you think.

### CODE REVIEW

**Comment by Mottie** [comment]
Hi @rayrutjes!

Sweet thanks! I've gotta run.. if no one gets around to merging this, I'll do it when I get a break later today.

**Comment by silverwind** [comment]
Thanks!

1) Does not work in Firefox for me, because the addon's styles outweigth our styles. Adding `!important` to every rule should fix that.
2) The addon sets `.awesome-autocomplete { border: none; }`, I'm not sure this is indended but I'd rather not have that border removed:



3) In your screenshot, there'a a black `x` icon on the search bar, can you make that brighter?

**Comment by Mottie** [comment]
Also, all instances of `#4183C4` need to be replaced with `/*[[base-color]]*/ #4183C4` so when the user selects a different base color, it will match.

It also looks like this definition is a tint of the base color... is there a way we could make it match the base color or use a more neutral color?

```css
form#search_form .awesome-autocomplete .twitter-typeahead .tt-input:focus {
  border-color: #51A7E8;
}
```

**Comment by rayrutjes** [comment]
Ok, working on it, thx

**Comment by silverwind** [comment]
> is there a way we could make it match the base color or use a more neutral color
I think one could add `filter: brightness(120%)` if it's text-only nodes like links.

**Comment by Mottie** [comment]
Oh, one more thing... I set my base-color to green and noticed these issues:

awesome complete

**Comment by rayrutjes** [comment]
Regarding the border: none, we will have to make an issue on the autocomplete repo I guess. Is a Firefox only issue.

**Comment by silverwind** [comment]
I wonder why it won't show in other browsers thought. [Here](https://github.com/algolia/github-awesome-autocomplete/blob/0017ca44361ca47c83a8b41ef3478fd1fe32c8ca/code/scss/content.sass#L70-L71) is the offending style.

**Comment by rayrutjes** [comment]
To be sure to not forget anything:
- [x] Allow for custom color replacements
- [ ] Simplify selectors
- [ ] Make input border visible
- [ ] Test in Firefox and add required "!important"
- [ ] .tt-input:focus neutral color

**Comment by rayrutjes** [comment]
Ok, I am good, now waiting for the border fix on AwesomeComplete's side.

File: github-dark.css [hunk 1]
```diff
@@ -2512,6 +2512,57 @@
...
+  }
+  .awesome-autocomplete .aa-query,
+  .awesome-autocomplete .aa-branding,
+  .awesome-autocomplete .aa-infos,
+  .awesome-autocomplete .aa-login,
+  .awesome-autocomplete .aa-company,
+  .awesome-autocomplete .aa-company i,
+  .awesome-autocomplete .aa-description,
+  .awesome-autocomplete .aa-issue-body{
+    color: #C0C0C0 !important;
```
Comment (silverwind) [inline github-dark.css]: Could you lowercase the letters in hex colors?

File: github-dark.css [hunk 2]
```diff
@@ -2512,6 +2512,57 @@
...
+  .awesome-autocomplete .aa-category {
+    background-color: #1D1D1D !important;
+    border-top: 1px solid #343434 !important;
+    border-bottom: 1px solid #343434 !important;
+    color: #A0A0A0 !important;
+  }
+  .awesome-autocomplete .tt-suggestion {
+    border-color: #343434 !important;
+  }
+  .awesome-autocomplete .tt-suggestion .octicon{
```
Comment (silverwind) [inline github-dark.css]: Please add a space before every `{`.

**Comment by Mottie** [comment]
I'm curious... the awesome complete dropdown is showing this repo as having 1230 stars, but the star/unstar button at the top of the page is showing 747. Why is there a huge difference?

**Comment by rayrutjes** [comment]
Oh my, you are so right, I have no idea about that gap, let's ping Algolia ^^.
I will make the other changes in a few minutes, fixing the border in firefox for now.

**Comment by Mottie** [comment]
I opened an issue: https://github.com/algolia/github-awesome-autocomplete/issues/40

**Comment by rayrutjes** [comment]
The only missing thing is the delete-icon, but my design skills are limited... There is no white version of the icon, do you have a fallback solution? Could we maybe use one of octotree's icon?

**Comment by Mottie** [comment]
I didn't see a delete icon anywhere, if it is solid black, then you can apply an invert filter:

```css
.delete-icon {
  -webkit-filter: invert(100%) !important;
          filter: invert(100%) !important;
}
```

**Comment by rayrutjes** [comment]
There you go!
image

File: github-dark.css [hunk 3]
```diff
@@ -2512,6 +2512,61 @@
...
+  .awesome-autocomplete .aa-category {
+    background-color: #1d1d1d !important;
+    border-top: 1px solid #343434 !important;
+    border-bottom: 1px solid #343434 !important;
+    color: #a0a0Aa !important;
+  }
+  .awesome-autocomplete .tt-suggestion {
+    border-color: #343434 !important;
+  }
+  .awesome-autocomplete .tt-suggestion .octicon {
```
Comment (Mottie) [inline github-dark.css]: Would you merge this definition with the one above? Everything else looks good to me! Thanks again!

Comment (rayrutjes) [inline github-dark.css]: You mean with the global definitions, those huge lists of classes? :kissing_closed_eyes:

Comment (rayrutjes) [inline github-dark.css]: Oh I get it now, only the octicon?

Comment (Mottie) [inline github-dark.css]: No, I meant line 2521-2522.

Comment (silverwind) [inline github-dark.css]: He probably means the one on line 2522.

Comment (rayrutjes) [inline github-dark.css]: @silverwind thx, it's done!

File: github-dark.css [hunk 4]
```diff
@@ -2512,6 +2512,59 @@
...
+  .awesome-autocomplete .aa-company,
+  .awesome-autocomplete .aa-company i,
+  .awesome-autocomplete .aa-description,
+  .awesome-autocomplete .aa-issue-body {
+    color: #c0c0c0 !important;
+  }
+  .awesome-autocomplete .aa-thumbnail ,
+  .awesome-autocomplete .aa-thumbnail {
+    background-color: #181818 !important;
+  }
```
Comment (silverwind) [inline github-dark.css]: Is this the user avatar background? If so, I'd make it slightly brighter to it stands out over the background. See the last user here (not sure why it fails to pull his avatar, he has one):

Comment (silverwind) [inline github-dark.css]: Oh, and the star should be grey, not colored.

File: github-dark.css [hunk 5]
```diff
@@ -2512,6 +2512,59 @@
...
+  .awesome-autocomplete .aa-branding,
+  .awesome-autocomplete .aa-infos,
+  .awesome-autocomplete .aa-login,
+  .awesome-autocomplete .aa-company,
+  .awesome-autocomplete .aa-company i,
+  .awesome-autocomplete .aa-description,
+  .awesome-autocomplete .aa-issue-body {
+    color: #c0c0c0 !important;
+  }
+  .awesome-autocomplete .aa-thumbnail ,
```
Comment (silverwind) [inline github-dark.css]: Unnecessary space before `,`

**Comment by rayrutjes** [comment]
Not sure about the thumbnail background, your decision.

**Comment by silverwind** [comment]
Well, it would affect any user with transparent backgrounds, which is not GitHub-like. Let's leave it as-is.

**Comment by silverwind** [comment]
Merged in https://github.com/StylishThemes/GitHub-Dark/commit/882b6341ff957bfc71c055e7bf3f7c420045d7bc, thanks!

**Pull request closed without merging**